### PR TITLE
feat(*): improve dependency resolution to use gradle

### PR DESCRIPTION
**Pull request by benmarten**
This PR includes:
- Updated the gradle template with center dependencies. Now the Leanplum Android SDK and Gson will be installed from maven repo. Added optional boilerplate lines for gcm, fcm, location.
- Updated build script to also include jars of gcm, fcm, location for old way of building unity apk.

### CODE REVIEW

File: LeanplumSample/Assets/Plugins/Android/mainTemplate.gradle [hunk 1]
```diff
@@ -1,51 +1,82 @@
...
-      }
-   }
+    repositories {
+        jcenter()
+        flatDir {
+            dirs 'libs'
+        }
+        maven {
+            url "https://repo.leanplum.com/"
+        }
```
Comment (Enuviel) [inline LeanplumSample/Assets/Plugins/Android/mainTemplate.gradle]: please add 
`        maven {
            url 'https://maven.google.com'
        }`

**Comment by MaitreyaBuddha** [comment]
LGTM

**Comment by benmarten** [comment]
@sayaansaha as discussed, please let me know if the gradle approach worked in your qa :) thanks

**Comment by alexisoyama** [comment]
@sayaansaha  @benmarten  What's the status here?

**Pull request merged**